src/Network/Response.php

```diff
@@ -538,13 +538,18 @@ protected function _sendHeader($name, $value = null)
 
     /**
      * Sends a content string to the client.
+     * If the content is a callable, it is invoked.
      *
      * @param string $content string to send as response body
      * @return void
      */
     protected function _sendContent($content)
     {
-        echo $content;
+        if (!is_string($content) && is_callable($content)) {
+            $content->__invoke();
```

Comment:
@ADmad I'm not totally following. In the case where you use a stream with a callback, it is the callbacks responsibility to actually echo the content out. You may even choose to use an output buffering function, like `flush()`, to ensure the data is sent right away. Again, this is similar to how HttpFoundation does this, which you can see [here](http://symfony.com/doc/current/components/http_foundation/introduction.html#streaming-a-response).